lib/cli/parse.rs

```diff
@@ -376,7 +377,7 @@ macro_rules! check_exactly_one_not_empty {
                 });
             }
         } else {
-            let mut non_empty_fields_with_values = vec![$((stringify!($x), $x)),+]
+            let mut non_empty_fields_with_values = [$((stringify!($x), $x)),+]
```

Comment:
Gotta admit,  I had to look this up. Not just the change, but the original code. So, this is subtle enough to deserve either rewriting into simpler code, or a comment.

I'd say the same about a complex regexp too. Any place where the average reader would do a double take deserves a comment.

Reply:
Done in [157832d](https://github.com/casper-ecosystem/casper-client-rs/pull/101/commits/157832dd2d6f4e4b85b952e8d456388064ec0206).